### DISCUSSION TITLE
Is there a way to programmatically setup & bootstrap a supabase instance?

### DISCUSSION THREAD

**Discussion by ziadomalikold** [supabase/supabase#29663] [Questions] [answered]
I want to make a template that I can use for all my future projects, however, there is a lot of repetitive work when it comes to:

- Settings
- Auth providers & respective credentials
- Extensions
- Email templates
- URL configurations
- RLS policies

It would be awesome if there was a way to programmatically edit all these values, so that I can run a script that will do it for me.

I have looked into the new [`supabase bootstrap`](https://supabase.com/blog/supabase-bootstrap#how-templates-work) command but there does not seem to be the ability to edit all of the above, but maybe I have overlooked something. I would appreciate some pointers / guidance in that direction. Thank you in advance.

**Comment by GaryAustin1** [comment, marked as answer]
These MIGHT be useful:
https://supabase.com/docs/reference/api/introduction
https://supabase.com/docs/reference/api/v1-update-auth-service-config

**Reply by ziadomalikold** [reply]
Thanks a lot, this seems like a great start.

- The run SQL query endpoint "should" solve the following problems already:
    - RLS policies
    - Extensions
- The update auth service config endpoint should fix all of the auth provider stuff.
- There are other endpoints that might be useful for the general project 'Settings', which is good.

As for Email templates, this seems to be the key? 

![image](https://github.com/user-attachments/assets/7425a719-d371-439d-83fb-e62c863d0575)

I will have to do some experimentation on how changing these values behave. 
What's left is URL configuration, I'll have to check how one could change that, other than that all is cool :3

**Comment by ziadomalikold** [comment]
@GaryAustin1 gave a very nice pointer, I will get to implementing a script sooner or later and then report back here. 

Not sure if the discussion should be closed now or I should close it after I provide feedback, 
I'll leave it open for good measure but feel free to just close it.
